yt_dlp/networking/_requests.py

```diff
@@ -0,0 +1,400 @@
+import contextlib
+import functools
+import http.client
+import logging
+import re
+import socket
+import warnings
+
+from ..dependencies import brotli, requests, urllib3
+from ..utils import bug_reports_message, int_or_none, variadic
+
+if requests is None:
+    raise ImportError('requests module is not installed')
+
+if urllib3 is None:
+    raise ImportError('urllib3 module is not installed')
+
+urllib3_version = tuple(int_or_none(x, default=0) for x in urllib3.__version__.split('.'))
+
+if urllib3_version < (1, 26, 17):
+    raise ImportError('Only urllib3 >= 1.26.17 is supported')
+
+if requests.__build__ < 0x023100:
+    raise ImportError('Only requests >= 2.31.0 is supported')
+
+from http.client import HTTPConnection
+
+import requests.adapters
+import requests.utils
+import urllib3.connection
+import urllib3.exceptions
```

Comment:
Are these still necessary after the above imports?

Reply:
It appears to work without, however my IDE complains about `Cannot find reference 'adapters' in '__init__.pyi'`  and auto complete & typing doesn't work for them :thinking:

Reply:
Some quick research that appears to be a pycharm thing. Think it is safe to exclude these then.

Reply:
ah ok nvm then